### ISSUE TITLE
Unsuccessful compilation of the library: problem with dlhandler_unix.cpp

### ISSUE THREAD

**Issue by axvutt** [openbabel/openbabel#1591]
- [x] I believe this to be a bug with Open Babel
- [ ] This is a feature request

## Environment Information
Open Babel version: 2-4-1
Operating system and version: Window 10, 64 bit

## Expected Behavior
Successful build of the Open Babel library.

## Actual Behavior
* cmake:
C:\OpenBabel\build>cmake ..\openbabel-2-4-1 -G"MinGW Makefiles" -DEIGEN2_INCLUDE_DIR=C:\Eigen-3.2.9 -DBUILD_GUI=OFF
--> Operation succeeded (did not find Eigen though...)

* mingw32-make:
C:\OpenBabel\build>mingw32-make UNICODE=0 BUILD=release SHARED=1 MONOLITHIC=1
--> Compilation error:

> C:\OpenBabel\openbabel-2-4-1\src\dlhandler_unix.cpp: In static member function 'static bool DLHandler::openLib(const string&)':
> C:\OpenBabel\openbabel-2-4-1\src\dlhandler_unix.cpp:156:36: error: cannot convert 'const char*' to 'LPCWSTR {aka const wchar_t*}' for argument '1' to 'HINSTANCE__* LoadLibraryW(LPCWSTR)'
>      if(LoadLibrary(lib_name.c_str()))
>                                     ^
> src\CMakeFiles\openbabel.dir\build.make:1387: recipe for target 'src/CMakeFiles/openbabel.dir/dlhandler_unix.obj' failed
> mingw32-make[2]: *** [src/CMakeFiles/openbabel.dir/dlhandler_unix.obj] Error 1
> CMakeFiles\Makefile2:1078: recipe for target 'src/CMakeFiles/openbabel.dir/all' failed
> mingw32-make[1]: *** [src/CMakeFiles/openbabel.dir/all] Error 2
> Makefile:139: recipe for target 'all' failed
> mingw32-make: *** [all] Error 2

IMHO, the error is caused by LoadLibrary() which is replaced by LoadLibraryW() when UNICODE is defined, whatever its value (see 'winbase.h', lines 2232, 2375 and 2582). Its argument, in 'dlhandler_unix.cpp' is not implicitely converted, hence the error.
Suggested fix: replace LoadLibrary() by LoadLibraryA().

**Comment by baoilleach** [comment]
Does it work if mingw32-make is used without any additional options?

**Comment by axvutt** [comment]
Yes, same error... `UNICODE` must be defined somewhere else by default. I just don't know where.
Command:
`C:\OpenBabel\build>mingw32-make -j4`
Display:
```
[ 19%] Building CXX object src/CMakeFiles/openbabel.dir/dlhandler_unix.obj
C:\OpenBabel\openbabel-2-4-1\src\dlhandler_unix.cpp: In static member function 'static bool DLHandler::openLib(const string&)':
C:\OpenBabel\openbabel-2-4-1\src\dlhandler_unix.cpp:155:36: error: cannot convert 'const char*' to 'LPCWSTR {aka const wchar_t*}' for argument '1' to 'HINSTANCE__* LoadLibraryW(LPCWSTR)'
     if(LoadLibrary(lib_name.c_str()))
                                    ^
src\CMakeFiles\openbabel.dir\build.make:1312: recipe for target 'src/CMakeFiles/openbabel.dir/dlhandler_unix.obj' failed
mingw32-make[2]: *** [src/CMakeFiles/openbabel.dir/dlhandler_unix.obj] Error 1
mingw32-make[2]: *** Waiting for unfinished jobs....
CMakeFiles\Makefile2:1078: recipe for target 'src/CMakeFiles/openbabel.dir/all' failed
mingw32-make[1]: *** [src/CMakeFiles/openbabel.dir/all] Error 2
Makefile:139: recipe for target 'all' failed
mingw32-make: *** [all] Error 2
```
Same thing if I retry without `-j4`

By googling a bit around, I found a few discussions regarding the (apparently recurrent) issue:
- https://stackoverflow.com/questions/1791578/how-do-i-convert-a-char-string-to-a-wchar-t-string
- https://stackoverflow.com/questions/14370982/converting-const-char-to-lpctstr-for-createdirectory
- and some similar others...

By looking at the declaration in `winbase.h`:
```
#ifdef UNICODE
: : :
#define LoadLibrary LoadLibraryW
: : :
#else
: : :
#define LoadLibrary LoadLibraryA
```
and the source code in `dlhandler_unix.cpp`:
```
#ifdef __MINGW32__
bool DLHandler :: openLib(const string& lib_name)
{
    if(LoadLibrary(lib_name.c_str()))
        return true;

    unsigned long err = GetLastError();
    return false;
}
#else
...
```
an idea could be whether to explicitely call `LoadLibraryA` or convert `lib_name.c_str()` to a `LPCWSTR` aka `const wchar_t*`. I tried the first option and the compilation succeeded. Now, I can't be sure this won't give problems in the future (I guess not...?).

**Comment by baoilleach** [comment]
Why don't you send a pull request and I can test it here on my machine.
